### ISSUE TITLE
Missing package: opencv-contrib-python

### ISSUE THREAD

**Issue by mflage** [piwheels/packages#299]
### Package name

opencv-contrib-python

### Package version

4.5.5.64

### PyPI URL

https://pypi.org/project/opencv-contrib-python/

### piwheels URL

https://www.piwheels.org/project/opencv-contrib-python/

### Python version

- [X] Python 3.5
- [X] Python 3.7
- [X] Python 3.9

### I am the maintainer

- [ ] Yes

### More information

Seems like the build fails for the latest version of opencv-contrib-python . The error seems related to a missing ninja build system.

**Comment by waveform80** [comment]
@bennuttall looks like the most recent opencv-contrib build failed. It's not a lack of the ninja build system (it fallsback successfully to GNU makefiles) but there's something else going on with cmake which, annoyingly, isn't getting included in the build log output. I'll take a look a bit later

**Comment by bennuttall** [comment]
Trying to build these on an 8GB Pi 4. Annoyingly it takes ages to fail. Will work my way through them on Bullseye and later try Bookworm.

**Comment by bennuttall** [comment]
I'm building these offline. It takes ages so please be patient. I'll keep this updated:

Version | cp37m | cp39 | cp311
-|-|-|-
4.8.1.78 | TODO | TODO | TODO
4.8.0.76 | TODO | TODO | TODO
4.8.0.74 | TODO | TODO | TODO
4.7.0.72 | TODO | TODO | TODO
4.7.0.68 | TODO | TODO | TODO
4.6.0.66 | TODO | TODO | TODO
4.5.5.64 | TODO | TODO | TODO

Please comment if another new version is released

**Comment by MichaIng** [comment]
Same as with `opencv-contrib-python-headless`: https://github.com/piwheels/packages/issues/282#issuecomment-3401637943
```
2025-09-04T12:42:03,506   virtual memory exhausted: Cannot allocate memory
2025-09-04T12:42:03,507   ninja: build stopped: subcommand failed.
```
On Bookworm instead:
```
2025-06-01T06:22:35,923     ERROR: Clock skew detected. File /tmp/pip-install-svf5ctsa/numpy_a8d2a72c9bca465b8f3833e89091e3bf/.mesonpy-zl_0gbu0/meson-python-native-file.ini has a time stamp 7631946.6341s in the future.
```
I first thought this was a random one-time issue, but since it happend for both `opencv-contrib-python` and `opencv-contrib-python-headless` on Bookworm/Python 3.11, maybe it is a general one. And indeed, the latest release was 2025-07-07, but the build log is from 2025-06-01, so the build system clock was over a month in the past for some reason 🤔.

**Comment by bennuttall** [comment]
I've fixed the clock skew issue so need to identify any builds with this error and rebuild them. I'll see what I can do.

**Comment by bennuttall** [comment]
I might have to spin up some 8GB Pi 4s and requeue all the opencv builds

**Comment by MichaIng** [comment]
Swap space does not work for this? I mean this makes it slow, but if it is needed for such a small number of packages only, with `vm.swappiness=1` usage of the swap space can be minimized.

Otherwise 16 GB RPi 5 😛. It's crazy how the price scales almost linearly with RAM size.

**Comment by bennuttall** [comment]
Yeah we don't have access to Pi 5 builders with Mythic (as Pi 5 doesn't support the PoE HAT yet I believe?)

We could build these offline using a Pi 5 but it's a faff.

**Comment by bennuttall** [comment]
I've imported some missing opencv wheels for cp313. Will look at cp311 when I get chance.

**Comment by bennuttall** [comment]
I've imported as many 3.11 and 3.13 wheels as I managed to build.
